fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexValidator.java

```diff
@@ -44,5 +46,49 @@ public void validate(@Nonnull MetaDataValidator metaDataValidator) {
         for (RecordType recordType : metaDataValidator.getRecordMetaData().recordTypesForIndex(index)) {
             LuceneIndexExpressions.validate(index.getRootExpression(), recordType.getDescriptor());
         }
+
+        validateIndexOptions(index);
+    }
+
+    @VisibleForTesting
+    public static void validateIndexOptions(@Nonnull Index index) {
+        validateAnalyzerNamePerFieldOption(LuceneIndexOptions.LUCENE_ANALYZER_NAME_PER_FIELD_OPTION, index);
+        validateAnalyzerNamePerFieldOption(LuceneIndexOptions.AUTO_COMPLETE_ANALYZER_NAME_PER_FIELD_OPTION, index);
+        validateAutoCompleteExcludedFields(index);
+    }
+
+    private static void validateAnalyzerNamePerFieldOption(@Nonnull String optionKey, @Nonnull Index index) {
+        String analyzerNamePerFieldOption = index.getOption(optionKey);
+        if (analyzerNamePerFieldOption != null) {
+            analyzerNamePerFieldOption = analyzerNamePerFieldOption.strip();
+            if (analyzerNamePerFieldOption.startsWith(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE)
+                    || analyzerNamePerFieldOption.startsWith(LuceneIndexOptions.DELIMITER_BETWEEN_ELEMENTS)
+                    || analyzerNamePerFieldOption.endsWith(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE)
+                    || analyzerNamePerFieldOption.endsWith(LuceneIndexOptions.DELIMITER_BETWEEN_ELEMENTS)) {
+                throw new MetaDataException("Index " + index.getName() + " has invalid " + optionKey + ": " + analyzerNamePerFieldOption);
+            }
```

Comment:
I think these checks are all extraneous, as they'd all get validated by the later check of each key-value pair.

Reply:
Good catch, removed these